pages/docs/integrations/langchain/typescript.mdx

```diff
@@ -71,97 +73,102 @@ const handler = new CallbackHandler({
 
 <Callout type="info" emoji="ℹ️">
   In short-lived environments (e.g. serverless functions), make sure to always
-  call `handler.shutdownAsync()` at the end to flush the queue and await all
+  call `langfuseHandler.shutdownAsync()` at the end to flush the queue and await all
   pending requests. ([Learn more](/docs/sdk/typescript#lambda))
 </Callout>
 
 #### Create a simple LLM call using Langchain
 
 ```typescript /callbacks: [langfuseHandler]/
-import { LLMChain } from "langchain/chains";
-import { OpenAI } from "langchain/llms/openai";
-import { PromptTemplate } from "langchain/prompts";
+import { PromptTemplate } from "@langchain/core/prompts";
+import { OpenAI } from "@langchain/openai";
 
 import { CallbackHandler } from "langfuse-langchain";
 
-// create a handler
-const handler = new CallbackHandler({
-  publicKey: LF_PUBLIC_KEY,
-  secretKey: LF_SECRET_KEY,
+// Create a callback handler
+const langfuseHandler = new CallbackHandler({
+  publicKey: LANGFUSE_PUBLIC_KEY,
+  secretKey: LANGFUSE_SECRET_KEY,
 });
 
-// create a model
+// Choose a model
 const model = new OpenAI({
   temperature: 0,
-  openAIApiKey: "sk-...",
+  openAIApiKey: "YOUR-API-KEY",
+  callbacks: [langfuseHandler] // Register your Langfuse callback in the constructor
 });
-// create a prompt
+
+// Create a prompt
 const prompt = PromptTemplate.fromTemplate(
   "What is a good name for a company that makes {product}?"
 );
-// create a chain
-const chain = new LLMChain({
-  llm: model,
-  prompt,
-  callbacks: [langfuseHandler],
-});
 
-// execute the chain
-const resA = await chain.call(
+// Create a chain and add the Langfuse callback handler
+const chain = prompt.pipe(model).withConfig({ callbacks: [langfuseHandler] }); // Register your Langfuse callback on chain creation
```

Comment:
Yes, we have had many users with missing events due to this. However, it might be that langchain improved this in the past months. I will create a ticket to check this and adjust the docs.